QA-328

tbeckham opened the pull request:
First pass at adding the ability to specify a custom eucalyptus cookbook with euca-deploy.

tbeckham committed:
Add support to specify a custom cookbook. Also, do a git pull to make sure we are up to date.

tbeckham committed:
Dont't need to determine cookbook name. We do a git clone and specify the target dir to ensure it's eucalyptus-cookbook. This results in less code changes.

viglesiasce commented on bin/euca-deploy:
```diff
@@ -14,13 +14,15 @@ if __name__ == '__main__':
     parser.add_argument('-e', '--environment', default='etc/environment.yml')
     parser.add_argument('-p', '--password', default='foobar')
     parser.add_argument('-b', '--branch', default='euca-4.1')
+    parser.add_argument('-C', '--cookbook', default='https://github.com/eucalyptus/eucalyptus-cookbook')
```
Id rather not have both -C and -c, also --cookbook is not descriptive. lets make it only --cookbook-repo, that way argparse will still work with --cookbook but the help will be more clear

viglesiasce commented on bin/euca-deploy:
```diff
@@ -14,13 +14,15 @@ if __name__ == '__main__':
                                            args.debug,
-                                           args.branch)
+                                           args.branch,
+                                           args.cookbook)
```
similarly change this to cookbook_repo

viglesiasce commented on eucadeploy/componentdeployer.py:
```diff
@@ -23,9 +24,10 @@ def __init__(self, password, environment_file='environment.yml',
+                  '{0} eucalyptus-cookbook;'
+                  'fi'.format(cookbook))
             local('cd eucalyptus-cookbook; git checkout {0};'.format(cookbook_branch))
+            local('git pull')
```
i dont believe this will run in the right dir but could be mistaken

viglesiasce replied:
We could use the fabric cd context manager: http://docs.fabfile.org/en/latest/api/core/context_managers.html#fabric.context_managers.cd

tbeckham replied:
Nice catch, I incorrectly assumed that changing dir in the previous execute local line would persist.

tbeckham committed:
Address comments from @viglesiasce
1. rename cookbook repo parameter to be less ambiguous
2. CD to cookbook repo dir that was we cloned to then do git pull

viglesiasce commented:
Looks good. Thanks @tbeckham.
